awx/ui_next/src/screens/Setting/AzureAD/AzureADEdit/AzureADEdit.jsx

```diff
@@ -1,25 +1,146 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
-import { withI18n } from '@lingui/react';
-import { t } from '@lingui/macro';
-import { Button } from '@patternfly/react-core';
-import { CardBody, CardActionsRow } from '../../../../components/Card';
-
-function AzureADEdit({ i18n }) {
+import React, { useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Formik } from 'formik';
+import { Form } from '@patternfly/react-core';
+import { CardBody } from '../../../../components/Card';
+import ContentError from '../../../../components/ContentError';
+import ContentLoading from '../../../../components/ContentLoading';
+import { FormSubmitError } from '../../../../components/FormField';
+import { FormColumnLayout } from '../../../../components/FormLayout';
+import { useSettings } from '../../../../contexts/Settings';
+import { RevertAllAlert, RevertFormActionGroup } from '../../shared';
+import {
+  EncryptedField,
+  InputField,
+  ObjectField,
+} from '../../shared/SharedFields';
+import useModal from '../../../../util/useModal';
+import useRequest from '../../../../util/useRequest';
+import { SettingsAPI } from '../../../../api';
+
+function AzureADEdit() {
+  const history = useHistory();
+  const { isModalOpen, toggleModal, closeModal } = useModal();
+  const { PUT: options } = useSettings();
+
+  const { isLoading, error, request: fetchAzureAD, result: azure } = useRequest(
+    useCallback(async () => {
+      const { data } = await SettingsAPI.readCategory('azuread-oauth2');
+      const mergedData = {};
+      Object.keys(data).forEach(key => {
+        if (!options[key]) {
+          return;
+        }
+        mergedData[key] = options[key];
+        mergedData[key].value = data[key];
+      });
+      return mergedData;
+    }, [options]),
+    null
+  );
+
+  useEffect(() => {
+    fetchAzureAD();
+  }, [fetchAzureAD]);
+
+  const {
+    error: submitError,
+    request: submitForm,
+    result: submitResult,
+  } = useRequest(
+    useCallback(async values => {
+      const result = await SettingsAPI.updateAll(values);
+      return result;
+    }, []),
+    null
+  );
+
+  useEffect(() => {
+    if (submitResult) {
+      history.push('/settings/azure/details');
```

Comment:
Curious why you are navigating this way instead of doing the navigation inside of the useRequest [here](https://github.com/ansible/awx/pull/8472/files#diff-eb84bb28b0549e1934e10a2c5de60ac40f5224fccd28a4a414fa0813c65b0c8bR50).  Is it possible that a submitResult not be returned?

Reply:
Both patterns achieve the same goal. It is possible that a submitResult not be returned if there is an error or if the request function does not return a result. I've seen both patterns used in our codebase, but I'm open to moving navigation into the request function, for brevity.